admin-dev/themes/new-theme/js/components/form/form-object-mapper.ts

```diff
@@ -369,6 +377,30 @@ export default class FormObjectMapper {
     }
   }
 
+  /**
+   * We need a custom checking method for equality, we don't use strict equality on purpose because it would result
+   * into a potential infinite loop if type doesn't match, which can easily happen with a number value in a text input.
+   *
+   * And we also try to see if both values have the same Number value, this avoids forcing a number input value when
+   * it's not written exactly the same way (like pending zeros)
+   *
+   * @param $input
+   * @param value
+   * @private
+   */
+  private hasSameValue(
+    $input: JQuery,
+    value: string | number | string[] | undefined,
+  ): boolean {
+    // eslint-disable-next-line eqeqeq
+    if (Number(numberCommaTransform(value)) == numberCommaTransform($input.val())) {
+      return true;
+    }
+
+    // eslint-disable-next-line eqeqeq
+    return value == $input.val();
+  }
```

Comment:
It's unclear to me what this function is meant to do, but it's is mixing up view management (loading a field's value with jquery) with business logic. You should extract the `.val()` from and pass it as a parameter.
```suggestion
  private areEqual(
    inputValue: string,
    referenceValue: string | number | string[] | undefined,
  ): boolean {
    // eslint-disable-next-line eqeqeq
    if (Number(numberCommaTransform(referenceValue)) == numberCommaTransform(inputValue)) {
      return true;
    }

    // eslint-disable-next-line eqeqeq
    return referenceValue == inputValue;
  }
```

Reply:
Yep, good idea